raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java

```diff
@@ -0,0 +1,510 @@
+        verify(view).onPaymentError(anyString());
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseNoAuth(), generateRandomString());
+        verify(view).onChargeCardSuccessful(any(ChargeResponse.class));
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onPinAuthModelSuggested_showOTP() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(PIN), generateRandomString());
+        verify(view).showOTPLayout(anyString(), anyString());
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODE() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(AVS_VBVSECURECODE), generateRandomString());
+        verify(view).onAVSVBVSecureCodeModelUsed(anyString(), anyString());
+
+    }
+
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODES() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateRandomChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(unknownResCodemsg);
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onUnknownAuthModelUsed_showUnknownAuthModelUsed() {
```

Comment:
@mikelis135 what does "show unknown auth model used" mean? Rename this test accordingly